### PR TITLE
Make naming of 'getXxxById' consistent across the project

### PR DESCRIPTION
**Pull request by barroij**
PR remake from https://github.com/BabylonJS/Babylon.js/pull/10355

The idea of this PR is to use `Id` instead of 'ID'

### CODE REVIEW

File: gui/src/2D/controls/colorpicker.ts [hunk 1]
```diff
@@ -33,7 +33,7 @@ export class ColorPicker extends Control {
     private _s = 1;
     private _v = 1;
 
-    private _lastPointerDownID = -1;
+    private _lastPointerDownId = -1;
```
Comment (barroij) [inline gui/src/2D/controls/colorpicker.ts:36]: private => no backward-compatibilty issue

File: gui/src/2D/controls/sliders/baseSlider.ts [hunk 2]
```diff
@@ -20,7 +20,7 @@ export class BaseSlider extends Control {
     protected _displayThumb = true;
     private _step = 0;
 
-    private _lastPointerDownID = -1;
+    private _lastPointerDownId = -1;
```
Comment (barroij) [inline gui/src/2D/controls/sliders/baseSlider.ts:23]: private => no backward-compatibilty issue

File: guiEditor/src/components/propertyTab/propertyTabComponent.tsx [hunk 3]
```diff
@@ -183,12 +183,12 @@ export class PropertyTabComponent extends React.Component<IPropertyTabComponentP
     }
 
     loadFromSnippet() {
-        const snippedID = window.prompt("Please enter the snippet ID to use");
+        const snippedId = window.prompt("Please enter the snippet ID to use");
```
Comment (barroij) [inline guiEditor/src/components/propertyTab/propertyTabComponent.tsx:186]: local variable => no backward-compatibilty issue

File: guiEditor/src/diagram/workbench.tsx [hunk 4]
```diff
@@ -157,9 +157,9 @@ export class WorkbenchComponent extends React.Component<IWorkbenchComponentProps
         this.loadToEditor();
     }
 
-    async loadFromSnippet(snippedID: string) {
+    async loadFromSnippet(snippedId: string) {
```
Comment (barroij) [inline guiEditor/src/diagram/workbench.tsx:160]: local variable => no backward-compatibilty issue

File: inspector/src/components/actionTabs/tabs/propertyGrids/particleSystems/particleSystemPropertyGridComponent.tsx [hunk 5]
```diff
@@ -186,16 +186,16 @@ export class ParticleSystemPropertyGridComponent extends React.Component<IPartic
         const scene = system.getScene()!;
         let isGpu = system instanceof GPUParticleSystem;
 
-        let snippedID = window.prompt("Please enter the snippet ID to use");
+        let snippedId = window.prompt("Please enter the snippet ID to use");
```
Comment (barroij) [inline inspector/src/components/actionTabs/tabs/propertyGrids/particleSystems/particleSystemPropertyGridComponent.tsx:189]: local variable => no backward-compatibilty issue

File: inspector/src/components/actionTabs/tabs/propertyGrids/sprites/spriteManagerPropertyGridComponent.tsx [hunk 6]
```diff
@@ -78,16 +78,16 @@ export class SpriteManagerPropertyGridComponent extends React.Component<ISpriteM
         const spriteManager = this.props.spriteManager;
         const scene = spriteManager.scene;
 
-        let snippedID = window.prompt("Please enter the snippet ID to use");
+        let snippedId = window.prompt("Please enter the snippet ID to use");
```
Comment (barroij) [inline inspector/src/components/actionTabs/tabs/propertyGrids/sprites/spriteManagerPropertyGridComponent.tsx:81]: local variable => no backward-compatibilty issue

File: loaders/src/glTF/1.0/glTFLoader.ts [hunk 7]
```diff
@@ -198,15 +198,15 @@ var loadAnimations = (gltfRuntime: IGLTFRuntime) => {
             var bufferInput = GLTFUtils.GetBufferFromAccessor(gltfRuntime, gltfRuntime.accessors[inputData]);
             var bufferOutput = GLTFUtils.GetBufferFromAccessor(gltfRuntime, gltfRuntime.accessors[outputData]);
 
-            var targetID = channel.target.id;
-            var targetNode: any = gltfRuntime.scene.getNodeByID(targetID);
+            var targetId = channel.target.id;
```
Comment (barroij) [inline loaders/src/glTF/1.0/glTFLoader.ts:201]: local variable => no backward-compatibilty issue

File: loaders/src/glTF/1.0/glTFLoader.ts [hunk 8]
```diff
@@ -626,8 +626,8 @@ var importMesh = (gltfRuntime: IGLTFRuntime, node: IGLTFNode, meshes: string[],
     var indexCounts = new Array<number>();
 
     for (var meshIndex = 0; meshIndex < meshes.length; meshIndex++) {
-        var meshID = meshes[meshIndex];
-        var mesh: IGLTFMesh = gltfRuntime.meshes[meshID];
+        var meshId = meshes[meshIndex];
```
Comment (barroij) [inline loaders/src/glTF/1.0/glTFLoader.ts:629]: local variable => no backward-compatibilty issue

File: loaders/src/glTF/1.0/glTFLoader.ts [hunk 9]
```diff
@@ -762,8 +762,8 @@ var importMesh = (gltfRuntime: IGLTFRuntime, node: IGLTFNode, meshes: string[],
     newMesh.subMeshes = [];
     var index = 0;
     for (var meshIndex = 0; meshIndex < meshes.length; meshIndex++) {
-        var meshID = meshes[meshIndex];
-        var mesh: IGLTFMesh = gltfRuntime.meshes[meshID];
+        var meshId = meshes[meshIndex];
```
Comment (barroij) [inline loaders/src/glTF/1.0/glTFLoader.ts:765]: local variable => no backward-compatibilty issue

File: nodeEditor/src/components/propertyTab/propertyTabComponent.tsx [hunk 10]
```diff
@@ -250,15 +250,15 @@ export class PropertyTabComponent extends React.Component<IPropertyTabComponentP
         const material = this.props.globalState.nodeMaterial;
         const scene = material.getScene();
 
-        let snippedID = window.prompt("Please enter the snippet ID to use");
+        let snippedId = window.prompt("Please enter the snippet ID to use");
```
Comment (barroij) [inline nodeEditor/src/components/propertyTab/propertyTabComponent.tsx:253]: local variable => no backward-compatibilty issue

File: src/Behaviors/Meshes/pointerDragBehavior.ts [hunk 11]
```diff
@@ -15,7 +15,7 @@ import "../../Meshes/Builders/planeBuilder";
  * A behavior that when attached to a mesh will allow the mesh to be dragged around the screen based on pointer events
  */
 export class PointerDragBehavior implements Behavior<AbstractMesh> {
-    private static _AnyMouseID = -2;
+    private static _AnyMouseId = -2;
```
Comment (barroij) [inline src/Behaviors/Meshes/pointerDragBehavior.ts:18]: private prop => no backward-compatibilty issue

File: src/Behaviors/Meshes/pointerDragBehavior.ts [hunk 12]
```diff
@@ -34,10 +34,18 @@ export class PointerDragBehavior implements Behavior<AbstractMesh> {
...
+    }
+    /** @deprecated Please use currentDraggingPointerId instead */
+    public set currentDraggingPointerID(currentDraggingPointerID: number) {
+        this.currentDraggingPointerId = currentDraggingPointerID;
+    }
     /**
      * The id of the pointer that is currently interacting with the behavior (-1 when no pointer is active)
      */
-    public currentDraggingPointerID = -1;
+    public currentDraggingPointerId = -1;
```
Comment (barroij) [inline src/Behaviors/Meshes/pointerDragBehavior.ts:48]: Comes with `currentDraggingPointerID`getter/setter for backward compatibility

File: src/Behaviors/Meshes/sixDofDragBehavior.ts [hunk 13]
```diff
@@ -40,10 +40,18 @@ export class SixDofDragBehavior implements Behavior<Mesh> {
...
+    }
+    /** @deprecated Please use currentDraggingPointerId instead */
+    public set currentDraggingPointerID(currentDraggingPointerID: number) {
+        this.currentDraggingPointerId = currentDraggingPointerID;
+    }
     /**
      * The id of the pointer that is currently interacting with the behavior (-1 when no pointer is active)
      */
-    public currentDraggingPointerID = -1;
+    public currentDraggingPointerId = -1;
```
Comment (barroij) [inline src/Behaviors/Meshes/sixDofDragBehavior.ts:54]: Comes with `currentDraggingPointerID`getter/setter for backward compatibility

File: src/LensFlares/lensFlareSystemSceneComponent.ts [hunk 14]
```diff
@@ -50,12 +50,15 @@ declare module "../abstractScene" {
...
+        /** @deprecated Please use getLensFlareSystemById instead */
+        getLensFlareSystemByID(id: string): Nullable<LensFlareSystem>;
+
         /**
          * Gets a lens flare system using its id
          * @param id defines the id to look for
          * @returns the lens flare system or null if not found
          */
-        getLensFlareSystemByID(id: string): Nullable<LensFlareSystem>;
+        getLensFlareSystemById(id: string): Nullable<LensFlareSystem>;
```
Comment (barroij) [inline src/LensFlares/lensFlareSystemSceneComponent.ts:61]: Comes with proxy function `getLensFlareSystemByID` for backward compatibility

File: src/Lights/Shadows/shadowGenerator.ts [hunk 15]
```diff
@@ -797,7 +797,7 @@ export class ShadowGenerator implements IShadowGenerator {
     protected _cachedPosition: Vector3 = new Vector3(Number.MAX_VALUE, Number.MAX_VALUE, Number.MAX_VALUE);
     protected _cachedDirection: Vector3 = new Vector3(Number.MAX_VALUE, Number.MAX_VALUE, Number.MAX_VALUE);
     protected _cachedDefines: string;
-    protected _currentRenderID: number;
+    protected _currentRenderId: number;
```
Comment (barroij) [inline src/Lights/Shadows/shadowGenerator.ts:800]: protected variable => no backward-compatibilty issue except if this class is intended to be derived from by the user, which I have no idea. Your take @RaananW  ?

File: src/Meshes/mesh.ts [hunk 16]
```diff
@@ -2505,12 +2505,17 @@ export class Mesh extends AbstractMesh implements IGetSetVerticesData {
...
+        return this.setMaterialById(id);
+    }
+
     /**
      * Sets the mesh material by the material or multiMaterial `id` property
      * @param id is a string identifying the material or the multiMaterial
      * @returns the current mesh
      */
-    public setMaterialByID(id: string): Mesh {
+    public setMaterialById(id: string): Mesh {
```
Comment (barroij) [inline src/Meshes/mesh.ts:2518]: Comes with proxy function `setMaterialByID`  for backward compatibility

File: src/Meshes/subMesh.ts [hunk 17]
```diff
@@ -551,9 +551,9 @@ export class SubMesh implements ICullable {
         var intersectInfo: Nullable<IntersectionInfo> = null;
 
         // Triangles test
-        let faceID = -1;
+        let faceId = -1;
```
Comment (barroij) [inline src/Meshes/subMesh.ts:554]: local variable => no issue

File: src/Misc/virtualJoystick.ts [hunk 18]
```diff
@@ -112,7 +112,7 @@ export class VirtualJoystick {
     private _axisTargetedByUpAndDown: JoystickAxis;
     private _joystickSensibility: number;
     private _inversedSensibility: number;
-    private _joystickPointerID: number;
+    private _joystickPointerId: number;
```
Comment (barroij) [inline src/Misc/virtualJoystick.ts:115]: private prop => no issue

File: src/Particles/cloudPoint.ts [hunk 19]
```diff
@@ -198,11 +198,19 @@ export class CloudPoint {
...
+    /** @deprecated Please use groupId instead */
+    public set groupID(groupID: number) {
+        this.groupId = groupID;
+    }
     /**
      * The group id
      * @hidden
      */
-    public groupID: number;
+    public groupId: number;
```
Comment (barroij) [inline src/Particles/cloudPoint.ts:213]: Comes with `groupID` getter/setter for backward compatibility

File: src/Particles/solidParticle.ts [hunk 20]
```diff
@@ -284,11 +284,19 @@ export class SolidParticle {
...
+    /** @deprecated Please use shapeId instead */
+    public set shapeID(shapeID: number) {
+        this.shapeId = shapeID;
+    }
     /**
      * The shape id
      * @hidden
      */
-    public shapeID: number;
+    public shapeId: number;
```
Comment (barroij) [inline src/Particles/solidParticle.ts:299]: Comes with `shapeID` getter/setter for backward compatibility

File: src/scene.ts [hunk 21]
```diff
@@ -2893,7 +2948,7 @@ export class Scene extends AbstractScene implements IAnimatable, IClipPlanesHold
         return null;
     }
 
-    private _getGeometryByUniqueID(uniqueId: number): Nullable<Geometry> {
+    private _getGeometryByUniqueId(uniqueId: number): Nullable<Geometry> {
```
Comment (barroij) [inline src/scene.ts:2951]: private method => no issue

**Review by barroij — COMMENTED** [review]
@RaananW that should better. Sorry for the reboot, that's what happens when one tries to do too  many things at the same time

File: src/Lights/Shadows/shadowGenerator.ts (hunk 15 above)
Comment (RaananW) [reply src/Lights/Shadows/shadowGenerator.ts:800]: fine with me.

**Review by RaananW — APPROVED** [review]

**Comment by deltakosh** [comment]
There is a documentation check issue

**Comment by barroij** [comment]
yep @deltakosh @RaananW , it seems it is because I did not mention the return value in the jsDoc of the deprecated functions or getter/setter that have been added to ensure backward compatibility.  I don't think it is needed but I can add them if you think the opposite :)

**Comment by deltakosh** [comment]
You have to for the build to succeed :D

**Pull request merged**